platform-sdk/docs/proposals/state-lifecycle/proposal.md

```diff
@@ -0,0 +1,270 @@
+# Lifecycle Management for the State
+
+## Summary
+
+This proposal describes a possible implementation of state lifecycle management that could work equally well for both the Consensus Node and the Block Node.
+
+|      Metadata      |                 Entities                 |
+|--------------------|------------------------------------------|
+| Designers          | [@imalygin](https://github.com/imalygin) |
+| Functional Impacts | Consensus Node, Block Node               |
+
+## Purpose and Context
+
+There is an initiative to create self-contained State modules (`swirlds-state-api` and `swirlds-state-impl`) that have minimal dependencies. Specifically, these modules should not depend on `swirlds-platform-core`. As part of this initiative, a state lifecycle management mechanism will be implemented for use by both the Consensus Node and the Block Node.
+
+### Requirements
+
+- Introduce an interface representing state lifecycle management and its implementation.
+- Update the current State API to align with the State Lifecycle API.
+- Implement these APIs and migrate existing code to use them.
+
+## Design & Architecture
+
+### Summary
+
+The following design describes a solution for managing the lifecycle of the state object. It covers the following aspects:
+
+- State initialization (loading a snapshot or creating a new state)
+- Maintaining references to the mutable state and the latest immutable state
+- Restricting mutability to a single state object
+- Loading snapshots from disk
```

Comment:
This is a repetition of the first bullet point. Should this rather be persistence a snapshot to disk?
I think serializing a snapshot should be a feature